lib/deadpool-cyclone/src/lib.rs

```diff
@@ -155,4 +158,55 @@ mod tests {
 
         instance.terminate().await.expect("failed to terminate");
     }
+
+    #[tokio::test]
+    async fn pow() {
+        let mut config_file = veritech_server::ConfigFile::default_local_uds();
+        veritech_server::detect_and_configure_development(&mut config_file)
+            .expect("failed to determine test configuration");
+
+        let spec = LocalUdsInstance::spec()
+            .try_cyclone_cmd_path(config_file.cyclone.cyclone_cmd_path())
+            .expect("failed to find cyclone program")
+            .cyclone_decryption_key_path(config_file.cyclone.cyclone_decryption_key_path())
+            .try_lang_server_cmd_path(config_file.cyclone.lang_server_cmd_path())
+            .expect("failed to find lang server program")
+            .limit_requests(2)
+            .runtime_strategy(LocalUdsRuntimeStrategy::LocalDocker)
+            .socket_strategy(LocalUdsSocketStrategy::Random)
+            .ping()
+            .build()
+            .expect("failed to build spec");
+        let manager = Manager::new(spec);
+
+        let mut instance = managed::Manager::create(&manager)
+            .await
+            .expect("failed to create instance");
+        dbg!(&instance);
+
+        let status = instance
+            .liveness()
+            .await
+            .expect("failed to run liveness check");
+        assert_eq!(status, LivenessStatus::Ok);
+        instance.ensure_healthy().await.expect("failed healthy");
+
+        let status = instance
+            .readiness()
+            .await
+            .expect("failed to run readiness check");
+        assert_eq!(status, ReadinessStatus::Ready);
+        instance.ensure_healthy().await.expect("failed healthy");
+
+        instance
+            .execute_ping()
+            .await
+            .expect("failed execute ping")
+            .start()
+            .await
+            .expect("failed to start protocol");
+        instance.ensure_healthy().await.expect("failed healthy");
+
+        instance.terminate().await.expect("failed to terminate");
```

Comment:
Great test! One cool thing you could do is write an infallible wrapper around the test and then whenever it returns, you call terminate. Here's an example:

```rust
#[tokio::test]
async fn pow() {
    // Set up the instance here.
    // --snip--

    if let Err(e) = pow_inner().await {
        instance.terminate().await.expect("failed to terminate");
        panic!("{}", e);
    } 
}
```

This way, we pretty much always terminate. It's not foolproof, but it could prove useful.